src/pulselistener/pulselistener/listener.py

```diff
@@ -68,72 +63,62 @@ def __init__(self, configuration):
 
     async def build_consumer(self, *args, **kwargs):
         '''
-        Main consumer, running queued builds
+        Main consumer, running queued builds from the web server
         '''
+        assert self.web_queue is not None, 'Missing web server queue'
+
         while True:
-            await self.run_builds()
 
-            # Sleep just a bit between two runs
-            await asyncio.sleep(2)
+            # Process next build in queue
+            if not self.web_queue.empty():
+                build = self.web_queue.get_nowait()
+                await self.run_build(build)
+
+            # Sleep just a bit between builds
+            await asyncio.sleep(1)
```

Comment:
If we use aioprocessing, we could avoid the wait loop with the 1 second delay, and just await on queue.get.
Or, we could do something like this:
```
def wait_build():
    return self.web_queue.get()

while True:
    build = await loop.run_in_executor(None, wait_build)
    await self.run_build(build)
```